extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -103,10 +104,13 @@ public OpenCensusProtobufInputRowParser withParseSpec(ParseSpec parseSpec)
   @Override
   public List<InputRow> parseBatch(ByteBuffer input)
   {
+    SettableByteEntity<ByteEntity> settableByteEntity = new SettableByteEntity<>();
+    settableByteEntity.setEntity(new ByteEntity(input));
     return new OpenCensusProtobufReader(
         parseSpec.getDimensionsSpec(),
-        new ByteEntity(input),
+        settableByteEntity,
         metricDimension,
+        null,
```

Comment:
it's bit odd to pass null here, since this would likely break if the downstream code ever tried to use this to write a column with a null name.

It looks like there is actually a bug in the existing OpenCensusProtobufInputFormat here since it does not actually ever use the valueDimension field when decoding OpenCensus and instead hard-codes it as `value`

I think we can fix that and have OpenCensusProtobufReader honor the `valueDimension` setting so we don't have to pass `null` here. This should not break any backwards compatibility, since we do not set the `valueDimension` anywhere in our configs today, and the default for valueDimension is `value`

Reply:
https://confluentinc.atlassian.net/browse/OBSDATA-566